authenticate.js

```diff
@@ -33,6 +41,10 @@ const authenticateToken = function (username, hostname, token) {
 	}
 };
 
-module.exports.authenticateToken = authenticateToken;
-module.exports.validate = validate;
-module.exports.s3oPublicKeyPromise = s3oPublicKey({ promise: true });
+module.exports = (getPublicKey) => {
+	const validate = validateFactory(getPublicKey);
+	return {
+		authenticateToken: authenticateToken(getPublicKey, validate),
```

Comment:
name the factory function something slightly different? getTokenAuthenticator?

Reply:
Done!